### PR TITLE
fix: broken link in documentation to block support mechanism

### LINKED ISSUE #33778: Link broken in documentation `https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#styles`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Have you tried deactivating all plugins except Gutenberg?

- [X] I have tested with all plugins deactivated.

### Have you tried replicating the bug using a default theme e.g. Twenty Twenty?

- [X] I have tested with a default theme.

### Description

The link to “block support mechanism” in the top section of Styles is broken - > https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#styles

The link points to https://developer.wordpress.org/block-editor/how-to-guides/themes/block-api/block-supports.md which is not present.

### Step-by-step reproduction instructions

1. Go to https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#styles
2. Click link "block support mechanism"

### Expected Behavior

A working link.

### Current Behavior

A non-working link.

### Screenshots or screen recording (optional)

_No response_

### Code snippet (optional)

_No response_

### WordPress Information

_No response_

### Gutenberg Information

_No response_

### What browsers are you seeing the problem on?

_No response_

### Device Information

_No response_

### Operating System Information

_No response_

### PR DESCRIPTION
**Pull request by RhnSharma**
## Description

Fixes #33778
Fix a broken link to block support mechanism in `theme-json.md`.

Hi @Mamaduka, how does this look?
Let me know if this needs any changes.
Thanks

## Types of changes

 Bug fix (non-breaking change which fixes an issue) 

## Checklist:
- [ ] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: docs/how-to-guides/themes/theme-json.md [hunk 1]
```diff
@@ -595,7 +595,7 @@ The Gutenberg plugin extends the styles available from WordPress 5.8, so they ca
 The tabs below show WordPress 5.8 supported styles and the ones supported by the Gutenberg plugin.
 </div>
 
-Each block declares which style properties it exposes via the [block supports mechanism](../block-api/block-supports.md). The support declarations are used to automatically generate the UI controls for the block in the editor. Themes can use any style property via the `theme.json` for any block ― it's the theme's responsibility to verify that it works properly according to the block markup, etc.
+Each block declares which style properties it exposes via the [block supports mechanism](../../reference-guides/block-api/block-supports.md). The support declarations are used to automatically generate the UI controls for the block in the editor. Themes can use any style property via the `theme.json` for any block ― it's the theme's responsibility to verify that it works properly according to the block markup, etc.
```
Comment (Mamaduka) [inline docs/how-to-guides/themes/theme-json.md]: I think this is the URL structure used in documentations - `/docs/reference-guides/block-api/block-supports.md`.

Everything else looks good 👍

Comment (RhnSharma) [reply docs/how-to-guides/themes/theme-json.md]: I have made the change you suggested. Please take a look at it and let me know if this needs any changes.
Thanks

Comment (Mamaduka) [reply docs/how-to-guides/themes/theme-json.md]: Thanks for the quick update!

**Review by Mamaduka — APPROVED** [review]
Thanks, @RhnSharma

Changes look good to me 👍

**Pull request merged**